src/components/SideNavigation/NavigationItem/index.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+import { NavigationItemTypes, TNavigationLinkPropTypes } from './types'
+import './index.scss'
+import Icon from '../../Icon'
+import Badge from '../../Badge'
+
+const NavigationItem = (props: TNavigationLinkPropTypes) => {
+  const {
+    As,
+    type,
+    open,
+    iconName,
+    showBadge,
+    expandable = false,
+    showAction = false,
+    actionElm,
+    userImage,
+    active = false,
+    badgeContent,
+    Child
+  } = props
+
+  const [childOpen, setChildOpen] = useState(false)
+
+  return (
+    <>
+      <div
+        className={classNames('navigation--wrapper', `navigation--wrapper--${type}`)}
+        onClick={() => setChildOpen(!childOpen)}
```

Comment:
I think we need to prevent this event if navigation is not opened, otherwise we should show `arrow` icon